sdk/src/main/java/com/mapbox/maps/renderer/MapboxRenderThread.kt

```diff
@@ -253,15 +263,53 @@ internal class MapboxRenderThread : Choreographer.FrameCallback {
         widgetTextureRenderer.render(widgetRenderer.getTexture())
       }
     } else {
+      logE("KIRYLDD", "render start $frameTimeNanos")
       mapboxRenderer.render()
+      logE("KIRYLDD", "render end $frameTimeNanos")
     }
 
     // assuming render event queue holds user's runnables with OpenGL ES commands
     // it makes sense to execute them after drawing a map but before swapping buffers
     // **note** this queue also holds snapshot tasks
     drainQueue(renderEventQueue)
+    if (hasViewAnnotations) {
+      hasViewAnnotations = false
```

Comment:
Feels weird that after swapping buffers flag becomes false, I'd rename to `needViewAnnotationsSync` or smth